### PR TITLE
Add ternary functors/substr

### PR DESCRIPTION
**Pull request by b-scholz**

### CODE REVIEW

File: configure.ac [hunk 1]
```diff
@@ -18,6 +18,7 @@ AC_CANONICAL_HOST   # host_cpu, host_vendor, and host_os
 AM_INIT_AUTOMAKE([foreign subdir-objects 1.9 tar-pax])
 AM_MAINTAINER_MODE
 AC_CONFIG_HEADERS([config.h])
+AC_CONFIG_MACRO_DIRS([m4])
```
Comment (mmcgr) [inline configure.ac]: Was this an intentional change? As far as I know, where it is supported it should have identical behaviour to the following line.

**Comment by b-scholz** [comment]
On my system (MAC OS X) I got some warnings. If this is inconvenient on linux, let’s chuck it.
Bernhard.

**Comment by mmcgr** [comment]
That is odd behaviour, then.

There is no error from having both, though, and between them they silence warnings on more systems so let's just leave them both in.

**Comment by b-scholz** [comment]
As far as I understand: one is to guide the libtools to write some scripts in the m4 directory; the other one guides autoconf to pick up some scripts -- but I am not 100% sure about this behaviour.

**Pull request merged**